Sources/CvUnitAI.cpp

```diff
@@ -2103,7 +2101,7 @@ void CvUnitAI::AI_workerMove()
 	{
 		const int iWorkers = GET_PLAYER(getOwner()).AI_totalUnitAIs(UNITAI_WORKER);
 
-		if (iWorkers > 3 && iWorkers > GET_PLAYER(getOwner()).getNumCities()
+		if (iWorkers > 3 && iWorkers > 5 * GET_PLAYER(getOwner()).getNumCities()
```

Comment:
Since this is in the section where AI is in serious financial trouble, then it should not need 5 workers per city.

Reply:
what defines as "serious financial trouble?"

Reply:
That expenditure is 85% or more than than the entire gross income when at 100% tax rate (0% science, culture espionage)
But not always when that is the case either, there's more rules to it than just that, like if tax rate is more than 50% and the current treasury cannot fund you for X amount of turns at the current rate (X differ by gamespeed, 10 turns on normal gamespeed); i.e. not hit 0 gold treasury within X amount of turn.

It's more complex than that though, but it is pretty bad when that function call returns true.
Leader personality also influence at what threshold of bad it consider to be financial trouble.